libraries/AP_Scripting/examples/checksum_firmware.lua

```diff
@@ -0,0 +1,27 @@
+--[[
+   This script reads the firmware, calculates a checksum and emits a text message.
+
+   It does this 10 seconds after boot.  And any time millis() wraps.
+--]]
+
+local flash_file_path = "@SYS/flash.bin"
+
+function update()
+   -- wait until we've been booted 10 seconds before sending:
+   local now = millis()
+   if now < 10000 then
+      -- not time yet
+      return update, 100
+   end
+
+   local result = fs:crc32(flash_file_path)
+   if result == nil then
+      gcs:send_text(3, string.format("checksum_firmware error: checksumming failed"))
+   else
+      gcs:send_text(3, string.format("checksum_firmware: 0x%x", result:toint()))
+   end
+
+   return  -- this script is one-shot 
+end
+
+return update()
```

Comment:
No need for the loop, just return update, 10000

Reply:
Those would be somewhat different semantics; that would emit 10 seconds after a scripting restart rather than instantaneously, for example.

Not saying it's bad, just different and not what was agreed upon with the person who asked me to do it :-)

Reply:
Really? This would display instantaneously after scripting restart? I'm skeptical, are you sure?